js/components/higher-order/accessibleMenu.js

```diff
@@ -0,0 +1,116 @@
+var React = require('react');
+var ReactDOM = require('react-dom');
+var CONSTANTS = require('../../constants/constants');
+
+var AccessibleMenu = function(ComposedComponent) {
+  return React.createClass({
+
+    componentDidMount: function() {
+      this.onKeyDown = this.onKeyDown.bind(this);
+      this.menuDomElement = ReactDOM.findDOMNode(this.composedComponent);
+
+      if (this.menuDomElement) {
+        this.menuDomElement.addEventListener('keydown', this.onKeyDown);
+      }
+    },
+
+    componentWillUnmount: function() {
+      if (this.menuDomElement) {
+        this.menuDomElement.removeEventListener('keydown', this.onKeyDown);
+      }
+    },
+
+    /**
+     * Keydown event handler. Implements arrow key navigation for menu items.
+     * @private
+     * @param {event} event The keyboard event object.
+     */
+    onKeyDown: function(event) {
+      if (
+        !event.target ||
+        !event.target.hasAttribute(CONSTANTS.KEYBD_FOCUS_ID_ATTR)
+      ) {
+        return;
+      }
+
+      switch (event.key) {
+        case CONSTANTS.KEY_VALUES.ARROW_UP:
+        case CONSTANTS.KEY_VALUES.ARROW_LEFT:
+          event.preventDefault();
+          this.focusOnMenuItemSibling(event.target, false);
+          break;
+        case CONSTANTS.KEY_VALUES.ARROW_DOWN:
+        case CONSTANTS.KEY_VALUES.ARROW_RIGHT:
+          event.preventDefault();
+          this.focusOnMenuItemSibling(event.target, true);
+          break;
+        default:
+          break;
+      }
+    },
+
+    /**
+     * Finds the previous or next sibling of the given menu item and gives it focus.
+     * @private
+     * @param {Element} menuItem The menuItem element whose sibling we want to focus on.
+     * @param {Boolean} useNextSibling Choses the sibling next to menuItem when true and the previous one when false.
+     */
+    focusOnMenuItemSibling: function(menuItem, useNextSibling) {
+      var menuItemsList = [];
+      if (this.menuDomElement) {
+        menuItemsList = this.menuDomElement.querySelectorAll('[' + CONSTANTS.KEYBD_FOCUS_ID_ATTR + ']');
```

Comment:
Since this query's off of the menu dom element for this specific accessible menu, is it possible to focus on another accessible menu's elements? If not, is this expected behavior?

Reply:
So far yes, this is expected because the component assumes that the whole view will be a single menu. There is currently no use case for having multiple menus active or treating different sections of a view as separate menus.

Reply:
Got it.